# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webcodecs/VideoDecoder.cpp
# Commit: 1ec020b42625
# Full Hash: 1ec020b42625dba5da44b244027a97fc4ea8d623
# Author: Chun-Min Chang <chun.m.chang@gmail.com>
# Date: 2023-06-30 04:06:43
# Regressor Bug: 1831451
# File Overlap Count: 1
# Description:
#   Bug 1831451 - Allow constructing VideoFrame without pixel format r=padenot
#   
#   Depends on D171166
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D171518
# ==============================================================================

diff -r b49ae6bc356a -r 1ec020b42625 dom/media/webcodecs/VideoDecoder.cpp
--- a/dom/media/webcodecs/VideoDecoder.cpp	Thu Jun 29 22:33:13 2023 +0000
+++ b/dom/media/webcodecs/VideoDecoder.cpp	Thu Jun 29 22:33:13 2023 +0000
@@ -535,12 +535,10 @@
     // TODO: Calculate displaySize
   }
 
-  // TODO: Allow constructing VideoFrame without pixel format
-  return MakeRefPtr<VideoFrame>(
-      aGlobalObject, aData->mImage,
-      format ? format.ref() : VideoPixelFormat::EndGuard_,
-      aData->mImage->GetSize(), aData->mImage->GetPictureRect(), displaySize,
-      Some(aDuration), aTimestamp, aColorSpace);
+  return MakeRefPtr<VideoFrame>(aGlobalObject, aData->mImage, format,
+                                aData->mImage->GetSize(),
+                                aData->mImage->GetPictureRect(), displaySize,
+                                Some(aDuration), aTimestamp, aColorSpace);
 }
 
 static nsTArray<RefPtr<VideoFrame>> DecodedDataToVideoFrames(